### PR TITLE
BD layout and operations for onchain order error classification

### PR DESCRIPTION
**Pull request by josojo**
The first step of: https://github.com/cowprotocol/services/issues/676

The intention is to still index orders that had some errors during indexing. This is needed, as we still need the order data to later initiate any refunds and the users should still see the orders in their order overview.

The following errors could happen:

```
QuoteIDNotFound: This can not happen during launch.
ValidToTooFarInFuture: We said that we don't need to constrict a valid, as onchain orders are expensive to place. But still we might wanna do this in the future. During launch, the front-end will enforce valid_to < 180 minutes
NonAcceptedOrderClass: It is currently possible to create limit orders from on-chain events even if they are disabled at the API level. We can use this error, in case we run into this situation.
NotAllowedBuyToken: If a buy token turns into a bad token between quoting and order placement, we can deal with it here.
```

### API changes
It incorporates these API changes:
https://github.com/cowprotocol/services/pull/723/files#diff-ea5f6887f8360abaf702f3dcece07c3b2e87022458254f48bb2fb66adf15b95fR612

### CODE REVIEW

File: crates/model/src/order.rs [hunk 1]
```diff
@@ -493,6 +493,27 @@ pub struct EthflowData {
     pub is_refunded: bool,
 }
 
+#[serde_as]
+#[derive(Eq, PartialEq, Clone, Derivative, Deserialize, Serialize)]
+#[derivative(Debug)]
+#[serde(rename_all = "camelCase")]
```
Comment (vkgnosis) [inline crates/model/src/order.rs:499]:
```suggestion
#[derive(Debug, Eq, PartialEq, Clone, Derivative, Deserialize, Serialize)]
```
I think we don't need the extra attributes. The struct below needs camelCase but not the rest either.

**Review by vkgnosis — APPROVED** [review]

File: crates/model/src/order.rs (hunk 1 above)
Comment (nlordell) [reply crates/model/src/order.rs:499]: `camelCase` has an effect on how the enum names get serialized/deserialized (`QuoteIdNotFound` vs `quoteIdNotFound`) so I would argue needing `serde(rename_all = ...)` depends on what the expected output is.

Agree that `serde_as` and `derivative` aren't used here and can be dropped.

**Review by nlordell — APPROVED** [review]
LGTM, some nits with the `derive` statements.

**Review by fedgiac — APPROVED** [review]
Looks good to me.

**Review by fedgiac — APPROVED** [review]
Nice!

**Pull request merged**